File: I/AINT/22S1/CMPS1131/OOP/preview/main.cpp
#include <iostream>
using namespace std;

// Class without constructors
class book
{
  // properties
public:
  int pages;
  string color;
  string title;
  string pageSize = "Letter";
};

// Class **with** constructors
class newBook
{
  // properties
public:
  int pages;
  string color;
  string title;
  string pageSize;

  newBook(int pages, string color, string title, string pageSize = "Letter")
  {
    this->pages = pages; // value
    this->color = color;
    this->title = title;
    this->pageSize = pageSize;
  }
};

int main()
{
  // Creating objects using classes
  // Syntax: class objName
  book book0;
  book0.pages = 200;
  book0.color = "violet";
  book0.title = "Lord of the Rings";
  book0.pageSize = "Letter";

  // Constructors are easier ways to create objects
  newBook book2(200, "violet", "Lord of the Rings");
  book2.pages = 50;

  // Accessing classes
  cout << book0.pages << endl;
  cout << book2.pages << endl;
  return 0;
}

File: I/AINT/22S1/CMPS1131/OOP/preview/javascript/main.js
// Yet another object
let library = {};

// Raw Objects - Can't be done in C++
let book0 = {
  // properties
  pages: 100, // value
  color: "pink",
  title: "Mathematics, Volume One",
  pageSize: "Letter",
};

// Adding object to object
library = { ...library, book0: book0 };

// Classes - Classes make objects
// Entire thing is called an object

// Class without constructors
class book {
  // properties
  pages;
  color;
  title;
  pageSize = "Letter";
}

// Class **with** constructors
class newBook {
  // properties
  pages;
  color;
  title;
  pageSize;
  constructor(pages, color, title, pageSize = "Letter") {
    this.pages = pages; // value
    this.color = color;
    this.title = title;
    this.pageSize = pageSize;
  }
}

// Accessing an object's properties
// object__class.property
// console.log("One book\n", book0.pages);

// Using classes and objects
book2 = new book();
book2.pages = 200;
book2.color = "violet";
book2.title = "Lord of the Rings";
book2.pageSize = "Letter";
// console.log("Book Without Constructors\n", book2);

library = { ...library, book2 };

// Using classes and objects with constructors
let book3 = new newBook(200, "violet", "Lord of the Rings");
// console.log("Book With Constructors\n", book3);
library = { ...library, book3 };

// Printing out an entire object
console.log("\n\nBook Library\n", "==========\n", library);
